cmd/tailscale/cli/configure-kube.go

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+//go:build !ts_omit_kube
+
+package cli
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/peterbourgon/ff/v3/ffcli"
+	"golang.org/x/exp/slices"
+	"k8s.io/client-go/util/homedir"
+	"sigs.k8s.io/yaml"
+	"tailscale.com/ipn/ipnstate"
+)
+
+func init() {
+	configureCmd.Subcommands = append(configureCmd.Subcommands, configureKubeconfigCmd)
+}
+
+var configureKubeconfigCmd = &ffcli.Command{
+	Name:       "kubeconfig",
+	ShortHelp:  "Configure kubeconfig to use Tailscale",
+	ShortUsage: "kubeconfig <hostname-or-fqdn>",
+	LongHelp: strings.TrimSpace(`
+Run this command to configure your kubeconfig to use Tailscale for authentication to a Kubernetes cluster.
+
+The hostname argument should be set to the Tailscale hostname of the peer running as an auth proxy in the cluster.
+`),
+	FlagSet: (func() *flag.FlagSet {
+		fs := newFlagSet("kubeconfig")
+		return fs
+	})(),
+	Exec: runConfigureKubeconfig,
+}
+
+func runConfigureKubeconfig(ctx context.Context, args []string) error {
+	if len(args) != 1 {
+		return errors.New("unknown arguments")
+	}
+
+	st, err := localClient.Status(ctx)
+	if err != nil {
+		return err
+	}
+	if st.BackendState != "Running" {
+		return errors.New("Tailscale is not running")
+	}
+	targetFQDN := args[0]
+	if !strings.Contains(targetFQDN, ".") {
+		// If the targetFQDN doesn't contain a dot, it's a hostname and we need to
+		// append the current tailnet's MagicDNS suffix.
+		if st.CurrentTailnet == nil {
+			return errors.New("no current tailnet")
+		}
+		if st.CurrentTailnet.MagicDNSSuffix == "" {
+			return errors.New("no magic DNS suffix")
+		}
+		if !st.CurrentTailnet.MagicDNSEnabled {
+			return errors.New("magic DNS is not enabled")
+		}
+		targetFQDN = fmt.Sprintf("%s.%s.", targetFQDN, st.CurrentTailnet.MagicDNSSuffix)
+	} else if !strings.HasSuffix(targetFQDN, ".") {
+		targetFQDN = targetFQDN + "."
+	}
+
+	confPath := filepath.Join(homedir.HomeDir(), ".kube", "config")
```

Comment:
you want to stat it first to see if it exists? Or do you expect it to not exist commonly?

Reply:
crap, I was relying on the old behavior of the k8s library. Will fix

Reply:
Done